api/resources_portal/models/notification.py

```diff
@@ -0,0 +1,86 @@
+        ("TRANSFER_REQUESTED", "TRANSFER_REQUESTED"),
+    )
+
+    objects = models.Manager()
+
+    created_at = models.DateTimeField(auto_now_add=True)
+
+    notification_type = models.CharField(max_length=32, choices=NOTIFICATION_TYPES)
+    notified_user = models.ForeignKey(
+        User, blank=False, null=False, on_delete=models.CASCADE, related_name="notifications"
+    )
+    associated_user = models.ForeignKey(
+        User,
+        blank=False,
+        null=True,
+        on_delete=models.CASCADE,
+        related_name="associated_notifications",
+    )
+    associated_organization = models.ForeignKey(
+        Organization, blank=False, null=True, on_delete=models.CASCADE
+    )
+    associated_material = models.ForeignKey(
+        Material, blank=False, null=True, on_delete=models.CASCADE
+    )
+
+    @computed(models.EmailField(blank=False, null=False))
+    def email(self):
+        return self.notified_user.email
+
+    @computed(models.TextField(null=False, blank=False))
+    def message(self):
+
+        # This dict of lambdas is neccessary because not all notifications have a material or organization, so evaluating them all as a dict results in a NoneType error.
+        alert_messages = {
+            "ORG_REQUEST_CREATED": lambda: f"{self.associated_user.username} is requesting to join {self.associated_organization.name}.",
+            "ORG_INVITE_CREATED": lambda: f"{self.associated_user.username} has invited you to join {self.associated_organization.name}.",
+            "ORG_INVITE_ACCEPTED": lambda: f"{self.associated_user.username} has accepted your invitation to join {self.associated_organization.name}.",
+            "ORG_REQUEST_ACCEPTED": lambda: f"{self.associated_user.username} has accepted your request to join {self.associated_organization.name}.",
+            "ORG_INVITE_REJECTED": lambda: f"{self.associated_user.username} has rejected your request to join {self.associated_organization.name}.",
+            "ORG_REQUEST_REJECTED": lambda: f"{self.associated_user.username} has rejected your invitation to join {self.associated_organization.name}.",
+            "ORG_INVITE_INVALID": lambda: f"You no longer have permissions to add members to {self.associated_organization.name}. Your pending invitations have been canceled.",
+            "ORG_REQUEST_INVALID": lambda: f"{self.associated_user.username} no longer has permissions to add members to {self.associated_organization.name}. Please resubmit your request.",
+            "MTA_UPLOADED": lambda: f"{self.associated_user.username} uploaded an MTA for {self.associated_material.title}.",
+            "APPROVE_REQUESTS_PERM_GRANTED": lambda: f"{self.associated_user.username} granted you permission to approve material transfer requests in {self.associated_organization.name}.",
+            "TRANSFER_REQUESTED": lambda: f'{self.associated_user.username} requested transfer of "{self.associated_material.title}".',
```

Comment:
I don't understand why our linter isn't complaining about these lines being >100 characters... That's our project's limit. I know this block has been tricky to write well, but these lines are very very long.

I Think this might be a decent way to break these up: https://stackoverflow.com/a/54950733?